src/vcpkg/commands.install.cpp

```diff
@@ -1506,4 +1507,30 @@ namespace vcpkg
 
         get_global_metrics_collector().track_string(StringMetric::InstallPlan_1, specs_string);
     }
+
+    void check_visual_studio_prompt_toolset_version(const ActionPlan& action_plan) {
+#if defined(_WIN32)
+        const auto vscmd_vctools_version_env = get_environment_variable("VCToolsVersion");
+        if (vscmd_vctools_version_env)
+        {
+            for (auto&& action : action_plan.install_actions)
+            {
+                if (action.abi_info)
+                {
+                    const auto& abi_info = action.abi_info.value_or_exit(VCPKG_LINE_INFO);
+                    if (abi_info.toolset)
+                    {
+                        auto toolset = abi_info.toolset.value_or_exit(VCPKG_LINE_INFO);
+                        auto v = vscmd_vctools_version_env.value_or_exit(VCPKG_LINE_INFO);
+                        if (toolset.full_version != v)
```

Comment:
I'm not sure toolset version is sufficient, as toolset version doesn't identify the VS instance. Can we use something like the path to the selected `cl.exe`?